### ISSUE TITLE
Custom Mode - Xiaomi Vacuum S10+ - dreame.vacuum.r2211o

### ISSUE THREAD

**Issue by ojoaosemtil** [merdok/homebridge-miot#515]
Hello,

I have a Xiaomi Vacuum S10+ - dreame.vacuum.r2211o - and there’s an option where, in the Xiaomi app, I can define specific modes to each room (bedroom standard mode, office strong mode, etc).
It would be nice to have a switch on HomeKit to just start this mode.

I tried to give specific configurations to each room in the plug-in, but if the last cleaning mode was “turbo”, it will use it next time I start it in HomeKit, even with the room cleaning specifically telling to clean in standard mode, for example.

 Can you help?

Thank you!
João

**Comment by merdok** [comment]
Hi,

you would need to find out how to set the mode for the specific room cleaning and include that in the config. 
Other then that you can create an automation where you first set the cleaning mode and afterwards start the room cleaning. That should also do the trick.

**Comment by ojoaosemtil** [comment]
Yes, I can create an automation when cleaning by room, but I would like to clean the whole house with different configuration each, I followed the guide you have with “costumeclean” for the x10+ but it doesn’t work with this model, or I am doing it the wrong way ahah

**Comment by merdok** [comment]
This is the part which is not really documented in the miot spec and any manufacturer can define the configuration as they want for their device, hence the only way to know that is to either guess how it could look like or capture the data which is sent by the mi home app to the cloud and analyze the requests. 
Since this is not a miot spec issue nor do I have the mentioned device, I am not able to help unfortunately.
Did you try search online if maybe someone else already figured that out?

**Comment by ojoaosemtil** [comment]
I understand, I’m trying to find some info, if I end up being able to do something I’ll let you know!
Thank you!

**Comment by s3melle** [comment]
miiocli dreamevacuum --ip 192.168.0.123 --token 123 raw_command action '{
  "siid":4,
  "aiid":1,
  "in":[
    {"piid":1,"value":18},
    {"piid":10,"value":"{\"selects\":[[6,1,1,3,1],[10,1,1,3,1]]}"}
  ]
}'


/*
    room,
    clean_mode,
    suction_level,
    water_level,
    times
*/
